From 7f0a8b04e12b1f07b56811071abb236d6aa971ef Mon Sep 17 00:00:00 2001
From: Jonathan Kim <jonathan.kim@amd.com>
Date: Wed, 8 Jan 2020 18:09:28 -0500
Subject: [PATCH 1282/1453] drm/amdgpu: disable xgmi p-states on vega20

Temporarily disable SMU messaging to switch p-states in non-upstream
branch due to FW bug.  This should only affect Vega20 since no other
program does an SMU invoke through driver.

Change-Id: I611a9ffe5183c510d7fcc5bb5955a4a6fad57aba
Signed-off-by: Jonathan Kim <Jonathan.Kim@amd.com>
Reviewed-by: Felix.Kuehling <Felix.Kuehling@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_xgmi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_xgmi.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_xgmi.c
index 5cf920d9358b..a37e9d0edede 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_xgmi.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_xgmi.c
@@ -279,7 +279,7 @@ int amdgpu_xgmi_set_pstate(struct amdgpu_device *adev, int pstate)
 	bool update_hive_pstate = true;
 	bool is_high_pstate = pstate && adev->asic_type == CHIP_VEGA20;
 
-	if (!hive)
+	if (!hive || adev->asic_type == CHIP_VEGA20)
 		return 0;
 
 	mutex_lock(&hive->hive_lock);
-- 
2.17.1

